surfactant/database_manager/database_utils.py

```diff
@@ -16,6 +19,74 @@
 from requests.exceptions import RequestException
 
 
+class BaseDatabaseManager(ABC):
+    """Abstract base class for managing pattern databases."""
+
+    def __init__(self, pattern_key: str, pattern_file: str, source: str):
+        self.pattern_key = pattern_key
+        self.pattern_file = pattern_file
+        self.source = source
+        self.new_hash: Optional[str] = None
+        self.download_timestamp: Optional[datetime] = None
+        self._database: Optional[Dict[str, Any]] = None
+
+    @property
+    @abstractmethod
+    def data_dir(self) -> Path:
+        """Returns the base directory for storing database files."""
+        # No implementation needed for abstract methods.
```

Comment:
Maybe this should default to returning `return ConfigManager().get_data_dir_path() /`, then subclasses can call this method in the parent class to get them to roughly the right subfolder.